[Bug]: Importing xgboost without it being marked as mandatory dependency

### Describe the bug

In automl/model you previously had try/except logic around xgboost __version__ import, making it optional.
However, the import is now not protected by try/except ImportError. 
Because this is not a core dependency, loading the code causes an ImportError, forcing you to manually install xgboost (even though it may not be not needed)

### Steps to reproduce

1. Take a clean Python environment
2. Install pyautogen, which installs flaml as a dependency
3. Load ChatResult
```python
from autogen.agentchat.chat import ChatResult
```


### Model Used

N/A

### Expected Behavior

Either xgboost is in mandatory requirements or the import can fail without propagating the error up the stack

### Screenshots and logs

```
______________ ERROR collecting tests/aiagents/test_rag_agents.py ______________
ImportError while importing test module '/home/runner/work/msticpy/msticpy/tests/aiagents/test_rag_agents.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/aiagents/test_rag_agents.py:3: in <module>
    from msticpy.aiagents.rag_agents import find_rst_files
msticpy/aiagents/rag_agents.py:17: in <module>
    from autogen.agentchat.chat import ChatResult
/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/autogen/__init__.py:9: in <module>
    from .agentchat import *
/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/autogen/agentchat/__init__.py:8: in <module>
    from .assistant_agent import AssistantAgent
/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/autogen/agentchat/assistant_agent.py:11: in <module>
    from .conversable_agent import ConversableAgent
/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/autogen/agentchat/conversable_agent.py:25: in <module>
    from ..code_utils import (
/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/autogen/code_utils.py:23: in <module>
    from autogen import oai
/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/autogen/oai/__init__.py:8: in <module>
    from autogen.oai.client import ModelClient, OpenAIWrapper
/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/autogen/oai/client.py:15: in <module>
    from flaml.automl.logger import logger_formatter
/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/flaml/automl/__init__.py:1: in <module>
    from flaml.automl.automl import AutoML, size
/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/flaml/automl/automl.py:20: in <module>
    from flaml.automl.ml import huggingface_metric_to_mode, sklearn_metric_name_set, spark_metric_name_dict, train_estimator
/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/flaml/automl/ml.py:12: in <module>
    from flaml.automl.model import BaseEstimator, TransformersEstimator
/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/flaml/automl/model.py:26: in <module>
    from xgboost import __version__ as xgboost_version
E   ModuleNotFoundError: No module named 'xgboost'
```

### Additional Information

_No response_

--- a/flaml/automl/__init__.py
+++ b/flaml/automl/__init__.py
@@ -1,5 +1,9 @@
-from flaml.automl.automl import AutoML, size
 from flaml.automl.logger import logger_formatter
-from flaml.automl.state import AutoMLState, SearchState
 
-__all__ = ["AutoML", "AutoMLState", "SearchState", "logger_formatter", "size"]
+try:
+    from flaml.automl.automl import AutoML, size
+    from flaml.automl.state import AutoMLState, SearchState
+
+    __all__ = ["AutoML", "AutoMLState", "SearchState", "logger_formatter", "size"]
+except ImportError:
+    __all__ = ["logger_formatter"]

